aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -985,35 +933,35 @@ public String clientAssumeRoleSessionName() {
   }
 
   public String s3FileIoSseType() {
-    return s3FileIoSseType;
+    return s3FileIOProperties.s3FileIoSseType();
```

Comment:
similar to the comment about variables, we should also just mark these methods as deprecated, with javadoc saying which version will this be removed. Overall my thought is that we should not have nested properties. We will have fully separated properties classes, similar to the `XXXConfig` in Trino, that seems to be pretty neat.

Reply:
> Overall my thought is that we should not have nested properties

So, we don’t want to make any changes to the methods is it? If so, then I have to add back the class fields to AwsProperties.
Which one do we want?
```

public String s3FileIoSseType() {
    return s3FileIoSseType;
}
```
  OR

```
public String s3FileIoSseType() {
  return s3FileIOProperties.s3FileIoSseType();
}
```